Make dynamism code robust to NotImplementedException

bobrenjc93 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #148823

In prod many models have `@property` methods that raise
NotImplementedError. This PR updates our dynamism code to be more robust
to these types of models.

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

bobrenjc93 committed:
Make dynamism code robust to NotImplementedException

In prod many models have @property methods that raise
NotImplementedError. This PR updates our dynamism code to be more robust
to these types of models.

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Make dynamism code robust to NotImplementedException"


In prod many models have property methods that raise
NotImplementedError. This PR updates our dynamism code to be more robust
to these types of models.

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

bobrenjc93 committed:
Update on "Make dynamism code robust to NotImplementedException"


In prod many models have `property` methods that raise
NotImplementedError. This PR updates our dynamism code to be more robust
to these types of models.

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented on torch/fx/experimental/_dynamism.py:
```diff
@@ -29,14 +29,21 @@ def module_to_nested_dict(module: torch.nn.Module) -> dict[str, Any]:
-                and isinstance(attr_value, (int, float, torch.Tensor))
-                and type(attr_value) is not bool
+        try:
+            if not attr_name.startswith("_") and not callable(
```
why do we skip _?

bobrenjc93 replied:
hmm good question, I actually don't remember... let me remove it

laithsakka commented on torch/fx/experimental/_dynamism.py:
```diff
@@ -29,14 +29,21 @@ def module_to_nested_dict(module: torch.nn.Module) -> dict[str, Any]:
+                getattr(module, attr_name)
             ):
-                self_dict[attr_name] = attr_value
+                attr_value = getattr(module, attr_name)
```
I wonder where exaclty is the error raised? when you do getattr(module, attr_name) ?

bobrenjc93 replied:
It's a @property decorator that makes it so we invoke the funciton and raise when we access it

laithsakka reviewed (approved):
I wonder

bobrenjc93 commented:
@pytorchbot merge

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Sorry but this is breaking internally, see D71042206 for details. To validate your fixes internally before relanding, you can follow the instructions here: https://fburl.com/fixing-ghfirst-reverts"

bobrenjc93 commented:
@pytorchbot rebase -b main

bobrenjc93 committed:
Update on "Make dynamism code robust to NotImplementedException"


In prod many models have `property` methods that raise
NotImplementedError. This PR updates our dynamism code to be more robust
to these types of models.

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

bobrenjc93 commented:
@bobrenjc93 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71208535).

bobrenjc93 commented:
Diff seems fine after rebasing: D71208535

bobrenjc93 commented:
@pytorchbot merge
